components/base/kinematicbase/ptgKinematics.go

```diff
@@ -160,7 +167,33 @@ func (ptgk *ptgBaseKinematics) Kinematics() referenceframe.Frame {
 func (ptgk *ptgBaseKinematics) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
 	ptgk.inputLock.RLock()
 	defer ptgk.inputLock.RUnlock()
-	return ptgk.currentInputs, nil
+	return ptgk.currentState.currentInputs, nil
+}
+
+func (ptgk *ptgBaseKinematics) ExecutionState(ctx context.Context) (motionplan.ExecutionState, error) {
+	if ptgk.Localizer == nil {
+		return motionplan.ExecutionState{}, errors.New("cannot call ExecutionState on a base without a localizer")
+	}
+	ptgk.inputLock.RLock()
+	currentIdx := ptgk.currentState.currentIdx
```

Comment:
This worries me slightly, I will have to track `currentState` across time to see if the following is a valid complaint.
I see that in `wrapWithPTGKinematics` we initialize our `currentState` via the `startingState` variable which only houses `currentInputs` information. 
Concretely, what worries me is that we could call `ExecutionState(ctx)` right after we have constructed our kinematic base.. This would then try to reference a `currentIdx` value which would end up causing a nil pointer dereference.

edit: ooo, perhaps there is some go thing that is going above my head where `ptgk.currentState.currentIdx` would just return `nil` opposed to crashing due to nil pointer dereference..

Reply:
`currentIdx` would not cause a nil pointer as it is not a pointer. It will return the default value for an int, which is 0.

Reply:
What about `currentExecutingSteps`?

Reply:
awesome, thank you for explaining this to me